### ISSUE TITLE
stuck in the 88 epoch

### ISSUE THREAD

**Issue by dk-liang** [facebookresearch/3detr#22]
When I train the sunrgbd, it always stuck in the 88 epoch without any information. Following the commands:
`sh scripts/sunrgbd_quick.sh`
![image](https://user-images.githubusercontent.com/23117298/147896572-89c11a9f-05e3-4233-85e5-6c558186723b.png)
![image](https://user-images.githubusercontent.com/23117298/147896589-a231f000-65e9-43c9-9858-0a2cdaf77a93.png)

**Comment by dk-liang** [comment]
When I use two GPUs for training, it will show the problem. But, if I utilize one GPU, it is ok. I do not know why.

**Comment by imisra** [comment]
My best guess is that a mini-batch with no ground truth is being sampled and since the outputs of this batch are not used to compute any loss, pytorch may hang.
Can you print `targets["num_boxes_replica"]` to verify this? If it is zero, then my guess might be true.

**Comment by sunmaosheng755** [comment]
Dear author, I have the same problem, I tried to print targets["num_boxes_replica"] and got 0， How can I solve it?

**Comment by sunmaosheng755** [comment]
> 我最好的猜测是正在对一个没有基本事实的小批量进行采样，并且由于该批量的输出不用于计算任何损失，因此 pytorch 可能会挂起。 您可以打印`targets["num_boxes_replica"]`以验证这一点吗？如果它为零，那么我的猜测可能是正确的。


Dear author, I have the same problem, I tried to print targets["num_boxes_replica"] and got 0， How can I solve it?

**Comment by sunmaosheng755** [comment]
> My best guess is that a mini-batch with no ground truth is being sampled and since the outputs of this batch are not used to compute any loss, pytorch may hang. Can you print `targets["num_boxes_replica"]` to verify this? If it is zero, then my guess might be true.

Dear author, I have the same problem, I tried to print targets["num_boxes_replica"] and got 0， How can I solve it?

**Comment by acheng1115** [comment]
> hi @dk-liang !

When I use two GPUs for training, it will show the same problem: it always stuck in the 88 epoch without any information.

Have you solved this problem? If so, how did you solve it？

Looking forward to hearing from you. Thanks!

**Comment by liuxiao916** [comment]
Same problem. It may caused by deadlock.

**Comment by yxchng** [comment]
how to solve this problem?
